src/libs/ajax/incremental-response/useIncrementalResponse.test.ts

```diff
@@ -44,18 +42,29 @@ describe('useIncrementalResponse', () => {
   });
 
   it('has loading state', async () => {
+    // Arrange
+    let getResponseController: PromiseController<IncrementalResponse<number>>;
+    const getControlledIncrementalResponse = () => {
+      const [promise, controller] = controlledPromise<IncrementalResponse<number>>();
+      getResponseController = controller;
+      return promise;
+    };
+
     // Act
-    const { result: hookReturnRef, waitForNextUpdate } = renderHook(() =>
-      useIncrementalResponse(getTestIncrementalResponse)
-    );
+    const { result: hookReturnRef } = renderHook(() => useIncrementalResponse(getControlledIncrementalResponse));
     const initialState = hookReturnRef.current.state;
-    await waitForNextUpdate();
+    await act(async () => {
+      getResponseController.resolve({
+        items: [1, 2, 3],
+        getNextPage: () => new Promise(() => {}),
+        hasNextPage: true,
+      });
+    });
 
     act(() => {
       hookReturnRef.current.loadNextPage();
     });
-    const stateAfterLoadingNextPage = hookReturnRef.current.state;
-    await waitForNextUpdate();
```

Comment:
You don't need to replace this `waitForNextUpdate`?

Reply:
The
```ts
act(() => {
  hookReturnRef.current.loadNextPage();
});
```
is enough to put it into the loading state, which is what this test is checking.